internal/engine/event_watch_manager.go

```diff
@@ -69,104 +89,115 @@ func (m *EventWatchManager) OnChange(ctx context.Context, st store.RStore) {
+// state, so we can match it later if the owner UID shows up.
+func (m *EventWatchManager) triageEventUpdate(event *v1.Event, objTree k8s.ObjectRefTree) model.ManifestName {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	uid := event.UID
+	oldEvent, ok := m.knownEvents[uid]
+
+	// Throw out updates that are older than what we currently have.
+	//
+	// Note that this code also dispatches actions for updates where the new
+	// ResourceVersion == the old ResourceVersion. We do this deliberately to make
+	// testing much easier, because the test harness doesn't need to keep track of
+	// ResourceVersions.
+	olderThanKnown := ok && oldEvent.ResourceVersion > event.ResourceVersion
+	if olderThanKnown {
+		return ""
 	}
 
-	mn := model.ManifestName(e.Labels()[k8s.ManifestNameLabel])
-	if mn == "" {
-		return ret
-	}
+	m.knownEvents[uid] = event
 
-	ret.obj = e
-	ret.manifest = mn
-	ret.belongsToThisTilt = e.Labels()[k8s.TiltRunIDLabel] == k8s.TiltRunID
-	return ret
-}
+	// Set up the descendent index of the involved object
+	for _, ownerUID := range objTree.UIDs() {
+		set, ok := m.knownDescendentEventUIDs[ownerUID]
+		if !ok {
+			set = store.NewUIDSet()
+			m.knownDescendentEventUIDs[ownerUID] = set
+		}
+		set.Add(uid)
+	}
 
-// This does not attempt to prevent duplicate simultaneous requests. If we get multiple events for the same
-// object at the same time, they can each result in their own API request.
-// We're currently assuming this matters sufficiently rarely that it's not worth the code complexity to fix.
-func (m *EventWatchManager) getEntry(ctx context.Context, obj v1.ObjectReference) uidMapEntry {
-	uid := obj.UID
-
-	m.uidMapMu.RLock()
-	entry, ok := m.uidMap[uid]
-	m.uidMapMu.RUnlock()
-	if !ok || entry.resourceVersion != obj.ResourceVersion {
-		entry = m.createEntry(ctx, obj)
-		m.uidMapMu.Lock()
-		// another thread might have come in and set this to the same or even a newer value by now
-		// neither of these cases should affect behavior aside from causing some unneeded api requests
-		m.uidMap[uid] = entry
-		m.uidMapMu.Unlock()
+	// Find the manifest name
+	for _, ownerUID := range objTree.UIDs() {
+		mn, ok := m.knownDeployedUIDs[ownerUID]
+		if ok {
+			return mn
+		}
 	}
 
-	return entry
+	return ""
 }
 
-// just loops and deletes any entrys that have hit their expiration
-func (m *EventWatchManager) uidMapJanitor(ctx context.Context) {
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		case <-m.clock.After(uidMapJanitorInterval):
-			m.uidMapMu.Lock()
-			for k, v := range m.uidMap {
-				if m.clock.Now().After(v.expiresAt) {
-					delete(m.uidMap, k)
-				}
-			}
-			m.uidMapMu.Unlock()
-		}
+func (m *EventWatchManager) dispatchEventChange(ctx context.Context, event *v1.Event, st store.RStore) {
+	objTree, err := m.ownerFetcher.OwnerTreeOfRef(ctx, event.InvolvedObject)
+	if err != nil {
+		logger.Get(ctx).Infof("Handling event update (%q): %v", event.Name, err)
```

Comment:
I'd s/Handling/Error handling/, in case `err` doesn't explicitly communicate to the user that there was an error.

Reply:
good point, done